app/gui2/src/util/ast/abstract.ts

```diff
@@ -885,6 +885,23 @@ export class Function extends Ast {
     this.equals_ = equals
     this.body_ = body
   }
+
+  static new(
+    module: MutableModule,
+    name: string,
+    args: Ast[],
+    exprs: Ast[],
+    id?: AstId | undefined,
```

Comment:
The `id` should not be assignable through the public constructor, similarly to all other AST classes.

Reply:
Done